core/src/core/template_engine_3

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+#
+# Author: Hans Chris Jones
+# Copyright 2018, LambdaStack
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# NOTE: You need to pass in the full paths to the file references below. The data is file should be private and it's
+# not in the project. The reason is you should create a YAML file that fits how you want to configure your
+# environment. For eample, you can have a small YAML data file for configuring the kickstart/ISO process and then
+# maybe one for building out the missing USER and/or SYSTEM data used in the CHEF Environment files. A sample
+# environment file ships with the project for vagrant called vagrant.json. However, a production.json should
+# really be a jinja2 template like base_environment.json.j2 with as much default data and with template {{ }} placeholders
+# for the actual data. The output of this process should be the TRUE production.json file. Also, it's a good idea
+# to name your production.json file more descriptive of the environment it actually belongs to. For example,
+# prod-dc101.json or something like it.
+
+import argparse
+import json
+import sys
+import traceback
+import logging
+
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+_SEPARATOR = '=' * 60
+
+
+# All three file paths must be full paths to each.
+def render_template(data_file, in_file, out_file, json_arg, yaml_arg):
+    template_dict = ""
+
+    logging.basicConfig(level=logging.INFO)
+    log = logging.getLogger(__name__)
+
+    # If the -j flag was passed then convert the yaml to pretty json in sorted order
+    if json_arg:
+        with open(data_file) as data:
+            template_dict = yaml.load(data)
+        log.info(json.dumps(template_dict, indent=4, sort_keys=True))
+        exit(0)

```

Comment:
```suggestion
        sys.exit(0)
```
see: https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python

Reply:
Changed according to documentation.

Reply:
Changed according to suggestion.